ui/app/components/sidebar/nav/cluster.hbs

```diff
@@ -40,28 +40,20 @@
     (and this.version.isEnterprise this.cluster.anyReplicationEnabled (has-permission "status" routeParams="replication"))
   }}
     <Nav.Title data-test-sidebar-nav-heading="Replication">Replication</Nav.Title>
-    {{#if this.cluster.dr.isSecondary}}
-      <Nav.Link
-        @route="vault.cluster.replication-dr-promote.details"
-        @text="DR Primary"
-        data-test-sidebar-nav-link="DR Primary"
-      />
-    {{else}}
```

Comment:
I doubled checked this and when the cluster is a secondary the sidebar isn't rendered since the user only has access to the dr route so this condition is no longer necessary.